addons/base_onboarding/models/base_onboarding_step.py

```diff
@@ -0,0 +1,75 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models
+from odoo.addons.base_onboarding.models.base_onboarding_state import ONBOARDING_STEP_STATES
+from odoo.tools.json import scriptsafe as json_safe
+
+
+class OnboardingStep(models.Model):
+    _name = 'base.onboarding.step'
+    _description = 'Onboarding Step'
+
+    name = fields.Char('Name of the onboarding step')
+    onboarding_id = fields.Many2one('base.onboarding', required=True)
```

Comment:
Would add ondelete='cascade' (as this is a submodel, should not prevent from removing onboardings)